### PR TITLE
UF-199: General support for UberFire navigation system. 

### PR DESCRIPTION
**Pull request by porcelli**
Now PerspectiveChange event keeps the original PlaceRequest. This is important for #136

### CODE REVIEW

**Comment by manstis** [comment]
@porcelli You need this merged before https://github.com/uberfire/uberfire/pull/136?

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/PerspectiveManager.java [hunk 1]
```diff
@@ -72,7 +75,8 @@ void switchToPerspective( final PerspectiveActivity perspective,
      * persisted perspective should load the Perspective definition from the applicable object.
      * @param doWhenFinished The command to execute once the perspective state have been removed. Must not be null.
      */
-    void removePerspectiveState( String perspectiveId, final Command doWhenFinished );
+    void removePerspectiveState( String perspectiveId,
+                                 final Command doWhenFinished );
```
Comment (manstis) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/PerspectiveManager.java]: Nice to split into 2 lines, but why not add ```final``` ;)?

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/PerspectiveManagerImpl.java [hunk 2]
```diff
@@ -72,8 +74,9 @@ public void loadPerspectiveStates( final ParameterizedCommand<Set<PerspectiveDef
     }
 
     @Override
-    public void removePerspectiveState( String perspectiveId, final Command doWhenFinished ) {
-        wbServices.removePerspectiveState(perspectiveId, doWhenFinished);
+    public void removePerspectiveState( String perspectiveId,
```
Comment (manstis) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/PerspectiveManagerImpl.java]: Ditto...

**Comment by manstis** [comment]
I don't see what this has to do with #136 ?

Looks like you're going to capture the ```PlaceRequest``` *after* the Perspective has been switched; I guess I was proposing to capture the ```PlaceRequest``` *before* the Perspective had been switched; e.g. if the Activity resolved in ```placeManager.goTo(...)``` was an AbstractPerspectiveActivity then keep this as the active perspective.. your approach is probably (possibly) better if the "new" perspective failed to be launched (IDK if we can veto a "goto"? - e.g. Editors preventing navigation if they contain dirty content?).

**Comment by porcelli** [comment]
Humm... I think now I got what you mean. It maybe over engineered, but it makes some sense. i'll try to talk to @cristianonicolai and clarify this with him.
Put on hold this for now ;)

**Comment by manstis** [comment]
@porcelli, open, closed, re-opened, closed, open, close.. make your mind up ;)

Did you want this merged?

**Comment by porcelli** [comment]
@manstis PR update.. mind merge it?

**Comment by manstis** [comment]
http://github.com/uberfire/uberfire/commit/5614a4051

**Pull request closed without merging**